src/Battlescape/Inventory.cpp

```diff
@@ -862,7 +865,16 @@ void Inventory::mouseClick(Action *action, State *state)
 								auto oldAmmo = item->setAmmoForSlot(slotAmmo, _selItem);
 								if (oldAmmo)
 								{
-									moveItem(oldAmmo, (item == weaponRightHand ? _inventorySlotLeftHand : _inventorySlotRightHand), 0, 0);
+									// If outside of battle and the reloaded weapon was on the ground, or the unit's hands were full, drop the ammo on the ground.
```

Comment:
If outside of battle and the weapon is in the backpack, the left hand is empty and the right hand contains a grenade... the old ammo after the reload ends up in the same slot (right hand) as the grenade... which is wrong and will cause issues.